### ISSUE TITLE
Add support for dynamic parameter interventions in Optimize

### ISSUE THREAD

**Issue by liunelson** [ciemss/pyciemss#582]
Optimize currently only support static parameter interventions. The dynamic variant appears to be necessary for the upcoming monthly demo. @sabinala  @anirban-chaudhuri is probably working on it already?

**Comment by Tom-Szendrey** [comment]
Discussed with @djinnome @sabinala and they have mentioned this is not required for this specific incoming scenario test case.
It sounds like this is being put on hold for the moment

**Comment by liunelson** [comment]
@sabinala @djinnome @anirban-chaudhuri 
As seen in today's meeting, we now have a Create Intervention Policy operator where one can define a policy with multiple different interventions (static/dynamic parameter/state). In case that the upcoming Monthly Demo does not require optimizing a dynamic parameter intervention, we would still need to have `pyciemss.optimize` accept the fixed form of all these interventions. Currently, there is only `fixed_static_parameter_interventions`. Could we expect also `fixed_*_interventions, * = static_state, dynamic_state, dynamic_parameter`?

https://github.com/ciemss/pyciemss/blob/422232cc31857765c3eb3365180c8d590599b5a9/pyciemss/interfaces.py#L786C5-L786C41

**Comment by anirban-chaudhuri** [comment]
@liunelson I can add in support for fixed dynamic interventions to `optimize`

**Comment by anirban-chaudhuri** [comment]
once PR #625 is merged, `optimize` can handle fixed interventions on:
- static parameter interventions (already existed)
- static state interventions
- dynamic parameter interventions
- dynamic state interventions
Note that the fixed dynamic interventions might compromise the optimizer's performance so extra care should be taken when using them.
